src/main/webapp/meal-page-script.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function fetchMealInfo() {
+    fetch('/meal-info').then(response => response.json()).then((meal) => 
+    {
+        const { title, description, ingredients } = meal;
+        const titleElement = document.getElementById("title");
+        titleElement.innerText = encodingCheck(title);
+        const descriptionElement = document.getElementById("description");
+        descriptionElement.innerText = encodingCheck(description);
+        const ingredientsElement = document.getElementById("ingredients");
+        for (const ingredient of ingredients) {
+            ingredientsElement.appendChild(createElementByTag(encodingCheck(ingredient), 'li'));
+        }
+        createMap();
+    });
+}
+
+function encodingCheck(string) {
```

Comment:
If it comes from Stack Overflow, it's often a good idea to include a link to the answer above the implementation.